### PR TITLE
Add spring gem to speed up rails and rspec

### PR DESCRIPTION
**Pull request by vassyz**
Adds the spring gem and related configuration and binstubs.

Specs take 6-7 seconds to load. We might want to investigate why it's so slow, but I think the addition of the spring gem is also a good improvement.

**With/without spring:**
Screenshot 2023-09-18 at 17 10 48

I have no idea why the total time becomes 0 with spring but I think most people don't need that. run the tests without it to get the accurate value.

### CODE REVIEW

File: config/environments/test.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
-  config.cache_classes = true
+  config.cache_classes = false
```
Comment (avinhurry) [inline config/environments/test.rb:10]: Could this slow the tests down when running the whole suite?

Comment (vassyz) [reply config/environments/test.rb:10]: They are marginally slower. I'll have a think.

File: bin/spring [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env ruby
+
+# This file loads Spring without using loading other gems in the Gemfile, in order to be fast.
```
[the remark this answers is not in the document]
Comment (vassyz) [reply bin/spring:3]: Thanks, this was auto-generated.

**Comment by vassyz** [comment]
@avinhurry made some changes so it only runs in development.

**Review by avinhurry — APPROVED** [review]
Looks good!

**Pull request merged**